### PR TITLE
Add static code checks to gitlab CD integration

### PR DESCRIPTION
**Pull request by zabbal**
The veracode pipeline is intended to be run as part of schedule only.
Test run: https://main.gitlab.in.here.com/olp/edge/ota/connect/back-end/ota-device-registry/pipelines/98127
Generated report: https://olp.pages.gitlab.in.here.com/edge/ota/connect/back-end/ota-device-registry/

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -3,6 +3,7 @@ stages:
   - test
```
Comment (simao) [inline .gitlab-ci.yml]: Please, there are a few things wrong with this PR

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -3,6 +3,7 @@ stages:
...
   - build image
   - trigger deploy
+  - publish
```
Comment (simao) [inline .gitlab-ci.yml]: publish is confusing in this context, someone looking at the pipelines would think this is publishing some jar, but it's actually just about deps scanning

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -78,3 +79,51 @@ trigger sit deploy:
   trigger:
     project: OLP/EDGE/OTA/infra/deployment-descriptors
     branch: master    
+
+
+trigger veracode scan:
```
Comment (simao) [inline .gitlab-ci.yml]: This is not triggering a scan. Trigerring in gitlab is to trigger some pipeline in other project, or in the same project but a different pipeline. See below for an example of a trigger.

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -78,3 +79,51 @@ trigger sit deploy:
...
+  # prepare and submit for static code analysis
+  stage: trigger deploy
```
Comment (simao) [inline .gitlab-ci.yml]: This is not trigerring a deploy, why is it on this stage? It's polluting the whole deploying pipeline

File: .gitlab-ci.yml [hunk 5]
```diff
@@ -78,3 +79,51 @@ trigger sit deploy:
...
+    - mv target/scala-*/device-registry_*.jar ./scan.jar
+  script:
+    - java -jar veracode-wrapper.jar -vid ${VERACODE_API_ID} -vkey ${VERACODE_API_KEY}
+      -action UploadAndScan -appname "OTA Backend - device registry" -createprofile true -autoscan true
+      -filepath scan.jar -version "job ${CI_JOB_ID} in pipeline ${CI_PIPELINE_ID} for ${CI_PROJECT_NAME} repo"
+  artifacts:
+    paths:
+      - scan.jar
+
+trigger-deps-scan:
```
Comment (simao) [inline .gitlab-ci.yml]: same comments as above

File: .gitlab-ci.yml [hunk 6]
```diff
@@ -78,3 +79,51 @@ trigger sit deploy:
...
+  # perform dependencies CVE analysis
+  stage: trigger deploy
```
Comment (simao) [inline .gitlab-ci.yml]: same as above

File: .gitlab-ci.yml [hunk 7]
```diff
@@ -78,3 +79,51 @@ trigger sit deploy:
...
+    - schedules
+  image: advancedtelematic/gitlab-jobs:0.1.0
+  script:
+    - ./sbt dependencyCheckAggregate
+    - mv target/scala-*/dependency-check-report.html ./depchk.html
+  artifacts:
+    paths:
+      - depchk.html
+
+pages:
```
Comment (simao) [inline .gitlab-ci.yml]: the name of the job is not descriptive of what it does.

File: .gitlab-ci.yml [hunk 8]
```diff
@@ -78,3 +79,51 @@ trigger sit deploy:
...
+  stage: publish
```
Comment (simao) [inline .gitlab-ci.yml]: why is this on this stage?

**Comment by simao** [comment]
Please revert, this is not ok

File: .gitlab-ci.yml (hunk 2 above)
Comment (zabbal) [reply .gitlab-ci.yml]: It's publishing index.html to github pages. Which alternative name would you suggest?

File: .gitlab-ci.yml (hunk 3 above)
Comment (zabbal) [reply .gitlab-ci.yml]: It's triggering scan in the veracode system. Which alternative name would you suggest?

File: .gitlab-ci.yml (hunk 4 above)
Comment (zabbal) [reply .gitlab-ci.yml]: Should I move it to another stage or make a new one?

File: .gitlab-ci.yml (hunk 5 above)
Comment (zabbal) [reply .gitlab-ci.yml]: There're several comments above - please clarify which you refer to exactly?

File: .gitlab-ci.yml (hunk 6 above)
Comment (zabbal) [reply .gitlab-ci.yml]: same as above

File: .gitlab-ci.yml (hunk 5 above)
Comment (zabbal) [reply .gitlab-ci.yml]: Which alternative name would you suggest?

File: .gitlab-ci.yml (hunk 7 above)
Comment (zabbal) [reply .gitlab-ci.yml]: Which alternative name would you suggest?

File: .gitlab-ci.yml (hunk 8 above)
Comment (zabbal) [reply .gitlab-ci.yml]: Should I move it to another stage or make a new one?

**Comment by zabbal** [comment]
> Please revert, this is not ok

I'd rather clarify first how exactly your comments should be addressed.

**Comment by simao** [comment]
I'd rather PRs are not merged without proper reviews.

**Comment by simao** [comment]
This is how a pipeline looks with this yml file:

pipeline

**Comment by zabbal** [comment]
And how should it look like?

File: .gitlab-ci.yml (hunk 2 above)
Comment (simao) [reply .gitlab-ci.yml]: "publish dependencies scan" for example

File: .gitlab-ci.yml (hunk 3 above)
Comment (simao) [reply .gitlab-ci.yml]: "start dependencies scan" or "start veracode scan" for example. Like I said, trigger in this context is something different.

File: .gitlab-ci.yml (hunk 4 above)
Comment (simao) [reply .gitlab-ci.yml]: make a new one

File: .gitlab-ci.yml (hunk 5 above)
Comment (simao) [reply .gitlab-ci.yml]: the name of the job

File: .gitlab-ci.yml (hunk 6 above)
Comment (simao) [reply .gitlab-ci.yml]: the stage where it's included.

File: .gitlab-ci.yml (hunk 7 above)
Comment (simao) [reply .gitlab-ci.yml]: I don't know what this job does so I cannot suggest a name.

File: .gitlab-ci.yml (hunk 8 above)
Comment (simao) [reply .gitlab-ci.yml]: I don't know whats this so I cannot say.

**Comment by zabbal** [comment]
See https://github.com/advancedtelematic/ota-device-registry/pull/113

**Pull request merged**